torch/utils/data/datapipes/iter/selecting.py

```diff
@@ -99,26 +93,25 @@ def _applyFilter(self, data, nesting_level):
 
     def _returnIfTrue(self, data):
         condition = self.filter_fn(data, *self.args, **self.kwargs)
-        if WITH_PANDAS:
-            if isinstance(condition, pandas.core.series.Series):
-                # We are operatring on DataFrames filter here
-                result = []
-                for idx, mask in enumerate(condition):
-                    if mask:
-                        result.append(data[idx:idx + 1])
-                if len(result):
-                    return pandas.concat(result)
-                else:
-                    return None
+
+        if df_wrapper.is_series_object(condition):
+            # We are operatring on DataFrames filter here
+            result = []
+            for idx, mask in enumerate(condition):
+                if mask:
+                    result.append(data[idx:idx + 1])
```

Comment:
should be `dataframe_wrapper.get_item(data, index=idx)`

Reply:
Done